js/runner.go

```diff
@@ -140,7 +173,15 @@ func (r *Runner) newVU(samplesOut chan<- stats.SampleContainer) (*VU, error) {
 			nameToCert[name] = &certs[i]
 		}
 	}
-
+	if r.Bundle.Options.Nic.ValueOrZero() != "" {
+		a := rr_getAddr(r.Bundle.Options.Nic.ValueOrZero())
+		if a != nil {
+			localTCPAddr := net.TCPAddr{
+				IP: a.IP,
+			}
+			r.BaseDialer.LocalAddr = &localTCPAddr
```

Comment:
This seems problematic... Take a look at the [k6 architecture](https://github.com/loadimpact/k6/wiki/Architecture) - the `Runner` (`r` here) is the thing that spawns VUs, so if you change the `LocalAddr` of its `BaseDialer` every time, you're just changing the `BaseDialer` all VUs would use. I think that with the current code all VUs will end up having the same `LocalAddr` configured, instead the thing that should be changed is probably the VU-local `Dialer` that's configured a few lines below this.